sdk/azcore/etag.go

```diff
@@ -0,0 +1,80 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"strings"
+)
+
+// ETag is a property used for optimistic concurrency during updates
```

Comment:
Is there a need to have a struct at all if it only has a single value? Have we considered just saying a '*string' is the etag and just exporting a compare function that takes two *string's? 

Or do you have some more stuff planned here?

Reply:
Good point, probably "type ETag *string" is good enough.

Reply:
Is there a reason it's `*string` and not just `string`?  The former would make it awkward for by-value cases.

Reply:
@jhendrixMSFT this allows the ETag to be nil which allows a differentiation of nil ETag and an ETag of `""`

Reply:
Yes. It is valid (although very unlikely) to have an empty-string etag. So, we need a way to say that is actually no etag at all.

Reply:
I see.  Is this common knowledge for ETag users, should we add a comment here explaining this?

Reply:
In the docs, we can reference this RFC: https://www.rfc-editor.org/rfc/rfc7232.html#section-2.3
The last example shows an empty string.